File: leet_BS_0035_insert_position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        L = 0
        R = len(nums) - 1
        while L <= R:
            i = (R - L) // 2 + L
            if nums[i] == target:
                return i
            elif nums[i] < target:
                L = i + 1
            else:
                R = i - 1
        return L

File: leet_BS_0035_insert_position.cc
class Solution {
public:
    int searchInsert(vector<int>& nums, int target)
    {
        int L = 0;
        int R = nums.size() - 1;
        while (L <= R)
        {
            int m = (R - L) / 2 + L;
            if (nums[m] == target)
                return m;
            else if (nums[m] > target)
                R = m - 1;
            else
                L = m + 1;
        }
        return L;
    }
};
